go-controller/pkg/ovn/namespace.go

```diff
@@ -365,8 +365,16 @@ func (oc *Controller) deleteNamespace(ns *kapi.Namespace) {
 
 	klog.V(5).Infof("Deleting Namespace's NetworkPolicy entities")
 	for _, np := range nsInfo.networkPolicies {
-		delete(nsInfo.networkPolicies, np.name)
-		oc.destroyNetworkPolicy(np, nsInfo)
+		oc.checkAndSkipRetryPolicy(np.policy)
+		// add the full np object to the retry entry, since the namespace is going to be removed
+		// along with any mappings of nsInfo -> network policies
+		oc.initRetryPolicyWithDelete(np.policy, np)
+		if err := oc.destroyNetworkPolicy(np, nsInfo); err != nil {
+			klog.Errorf("Failed to delete network policy: %s, error: %v", getPolicyNamespacedName(np.policy), err)
+			oc.unSkipRetryPolicy(np.policy)
+		} else {
+			delete(nsInfo.networkPolicies, np.name)
```

Comment:
@trozet do we need `oc.checkAndDeleteRetryPolicy(policy)` here on destroy success?

Reply:
yeah, nice catch. I think I will add an API to the cache to query if an entry exists in it. I can then use that in the unit test to detect if the cache is as expected during/after the test.